fabric-common/lib/BlockDecoder.js

```diff
@@ -564,428 +604,498 @@ rule
+			chaincode_id: filteredChaincodeActionProto.chaincode_event.chaincode_id,
+			tx_id: filteredChaincodeActionProto.chaincode_event.tx_id,
+			event_name: filteredChaincodeActionProto.chaincode_event.event_name
+			// filtered events do not have a payload
+		};
+	}
+
+	return chaincode_action;
+}
+
+function decodePrivateData(privateDataMapProto) {
+	if (!privateDataMapProto) {
+		logger.debug('decodePrivateData - private data is missing');
 		return {};
 	}
 	const private_data_map = {};
-	for (const txIndex in private_data_map_proto) {
-		const txPvtReadWriteSet = {};
-		const txPvtReadWriteSet_proto = private_data_map_proto[txIndex];
-
-		txPvtReadWriteSet.data_model = txPvtReadWriteSet_proto.data_model;
-		txPvtReadWriteSet.ns_pvt_rwset = [];
-		for (const ns_pvt_rwset_proto of txPvtReadWriteSet_proto.ns_pvt_rwset) {
+	let found = false;
+	// map key is the transaction index of the transaction for the private data
+	for (const txIndex in privateDataMapProto) {
+		const tx_pvt_read_write_set = {};
+		const txPvtReadWriteSetProto = privateDataMapProto[txIndex];
+
+		tx_pvt_read_write_set.data_model = txPvtReadWriteSetProto.data_model; // only KV=0
+		tx_pvt_read_write_set.ns_pvt_rwset = [];
+		for (const nsPvtRwsetProto of txPvtReadWriteSetProto.ns_pvt_rwset) {
 			const ns_pvt_rwset = {};
-			ns_pvt_rwset.namespace = ns_pvt_rwset_proto.namespace;
+			ns_pvt_rwset.namespace = nsPvtRwsetProto.namespace;
 			ns_pvt_rwset.collection_pvt_rwset = [];
-			for (const collection_pvt_rwset_proto of ns_pvt_rwset_proto.collection_pvt_rwset) {
+			for (const collectionPvtRwsetProto of nsPvtRwsetProto.collection_pvt_rwset) {
 				const collection_pvt_rwset = {};
-				collection_pvt_rwset.collection_name = collection_pvt_rwset_proto.collection_name;
-				collection_pvt_rwset.rwset = decodeKVRWSet(collection_pvt_rwset_proto.rwset);
+				collection_pvt_rwset.collection_name = collectionPvtRwsetProto.collection_name;
+				collection_pvt_rwset.rwset = decodeKVRWSet(collectionPvtRwsetProto.rwset);
 				ns_pvt_rwset.collection_pvt_rwset.push(collection_pvt_rwset);
 			}
-			txPvtReadWriteSet.ns_pvt_rwset.push(ns_pvt_rwset);
+			tx_pvt_read_write_set.ns_pvt_rwset.push(ns_pvt_rwset);
 		}
-		private_data_map[txIndex] = txPvtReadWriteSet;
+		const intIndex = Long.fromValue(txIndex).toInt();
+		private_data_map[intIndex] = tx_pvt_read_write_set;
+		found = true;
+	}
+
+	if (!found) {
+		logger.debug('decodePrivateData - no private data');
 	}
 
 	return private_data_map;
 }
 
-function decodeBlockHeader(proto_block_header) {
+function decodeBlockHeader(blockHeaderProto) {
 	const block_header = {};
-	block_header.number = proto_block_header.getNumber().toString();
-	block_header.previous_hash = proto_block_header.getPreviousHash().toBuffer().toString('hex');
-	block_header.data_hash = proto_block_header.getDataHash().toBuffer().toString('hex');
+	block_header.number = blockHeaderProto.number;
+	block_header.previous_hash = blockHeaderProto.previous_hash;
```

Comment:
Does it mean we deliver the responsibility to developer about how to visualize this?

Reply:
yes, we should not be making that decision, this way they get the field as defined in the protobuf